### ISSUE TITLE
A bit more input sanitation may be necessary.

### ISSUE THREAD

**Issue by Leopere** [nitnelave/lldap#239]
It would appear that the users name field is able to accept input with spaces and non alpha numeric characters.  It seems that when I did this the system seemed to lock up and slow down significantly.

**Comment by nitnelave** [comment]
There's fundamentally no limitation regarding what format usernames take, they're just case insensitive. The problem might be with downstream applications who will not recognize the string as a valid username.
I doubt that it affects performance though, we don't do anything smart with the username apart from lowercasing it.

One other thing to consider is that it's the admins who create users and thus choose the usernames, so I'm counting on them to be a bit reasonable ;)

I might forbid spaces though, just in case.

**Comment by olaxe** [comment]
I use emails as User ID because it is much more easy for my users to remember it. Is it possible to keep also authorized all email characters?

**Comment by nitnelave** [comment]
Note that you can configure most services to allow logging in via both email and username, in the LDAP settings

**Comment by olaxe** [comment]
Yes thanks. Unfortunately, not the LLDAP reset password process. My users are good at reset their passwords.
